### PR TITLE
Use different endpoint for custom workflow carrier account creation (FedEx, UPS)

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Carrier account creation now determines endpoint based on type

# Testing

- New unit test, cassette

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc.)

### CODE REVIEW

**Review by jchen293 — DISMISSED** [review]

**Review by Justintime50 — CHANGES_REQUESTED** [review]
We just pushed out the release candidate, why are we making big changes like moving all the constants around?

**Comment by nwithan8** [comment]
> We just pushed out the release candidate, why are we making big changes like moving all the constants around?

Mirroring the changes made when this feature was added to the C# library, we need to unify our constants under one file. Moved the exception-specific constants to the global constants file.

**Comment by Justintime50** [comment]
@nwithan8 please move the exception changes to a new PR so we can keep them small and focused. The point of this PR is to address carrier account registration only.

**Comment by nwithan8** [comment]
> @nwithan8 please move the exception changes to a new PR so we can keep them small and focused. The point of this PR is to address carrier account registration only.

https://github.com/EasyPost/easypost-java/pull/211

**Review by Justintime50 — APPROVED** [review]

File: src/main/java/com/easypost/service/CarrierAccountService.java [hunk 1]
```diff
@@ -30,13 +31,18 @@ public class CarrierAccountService {
      * @return created CarrierAccount object.
      * @throws EasyPostException when the request fails.
      */
-    public CarrierAccount create(final Map<String, Object> params)
-            throws EasyPostException {
+    public CarrierAccount create(final Map<String, Object> params) throws EasyPostException {
+        String type = (String) params.get("type");
+        if (type == null) {
+            throw new EasyPostException("Carrier account type is required");
```
Comment (jchen293) [inline src/main/java/com/easypost/service/CarrierAccountService.java:37]: You should use `MissingParameterError` here to be more explicit on the error type. There is a constant you can use `MISSING_REQUIRED_PARAMETER`

Comment (nwithan8) [reply src/main/java/com/easypost/service/CarrierAccountService.java]: Resolved

**Review by jchen293 — APPROVED** [review]

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**